apps/speed.c

```diff
@@ -3161,6 +3554,369 @@ int speed_main(int argc, char **argv)
+                goto kem_err_break;
+            }
+            if (EVP_PKEY_encapsulate(kem_encaps_ctx, out, &out_len,
+                                     send_secret, &send_secret_len) <= 0) {
+                BIO_printf(bio_err, "Encaps error for %s.\n", kem_name);
+                goto kem_err_break;
+            }
+            /* Now prepare decaps data structs */
+            kem_decaps_ctx = EVP_PKEY_CTX_new_from_pkey(app_get0_libctx(),
+                                                        pkey,
+                                                        app_get0_propq());
+            if ((kem_decaps_ctx == NULL)
+                || (EVP_PKEY_decapsulate_init(kem_decaps_ctx, NULL) <= 0) 
+                || ((strncmp(kem_name, "rsa", 3) == 0
+                  && (EVP_PKEY_CTX_set_kem_op(kem_decaps_ctx, "RSASVE") <= 0)))
+                || (((strncmp(kem_name, "EC", 2) == 0)
+                     || (strcmp(kem_name, "X25519") == 0)
+                     || (strcmp(kem_name, "X448") == 0))
+                  && (EVP_PKEY_CTX_set_kem_op(kem_decaps_ctx, "DHKEM") <= 0))
+                || (EVP_PKEY_decapsulate(kem_decaps_ctx, NULL, &rcv_secret_len,
+                                      out, out_len) <= 0)) {
+                BIO_printf(bio_err,
+                           "Error while initializing decaps data structs for %s.\n",
+                           kem_name);
+                goto kem_err_break;
+            }
+            rcv_secret = OPENSSL_malloc(rcv_secret_len);
+            if (rcv_secret == NULL) {
+                BIO_printf(bio_err, "MemAlloc failure in decaps for %s.\n",
+                           kem_name);
+                goto kem_err_break;
+            }
+            if ((EVP_PKEY_decapsulate(kem_decaps_ctx, rcv_secret,
+                                      &rcv_secret_len, out, out_len) <= 0)
+                || (rcv_secret_len != send_secret_len)
+                || (memcmp(send_secret, rcv_secret, send_secret_len))) {
+                BIO_printf(bio_err, "Decaps error for %s.\n", kem_name);
+                goto kem_err_break;
+            }
+            loopargs[i].kem_gen_ctx[testnum] = kem_gen_ctx;
+            loopargs[i].kem_encaps_ctx[testnum] = kem_encaps_ctx;
+            loopargs[i].kem_decaps_ctx[testnum] = kem_decaps_ctx;
+            loopargs[i].kem_out_len[testnum] = out_len;
+            loopargs[i].kem_secret_len[testnum] = send_secret_len;
+            loopargs[i].kem_out[testnum] = out;
+            loopargs[i].kem_send_secret[testnum] = send_secret;
+            loopargs[i].kem_rcv_secret[testnum] = rcv_secret;
+            break;
+
+            kem_err_break:
+            ERR_print_errors(bio_err);
+            op_count = 1;
+            kem_checks = 0;
+            break;
+
+        }
+        if (kem_checks != 0) {
+            kskey_print_message(kem_name, "keygen", kems_c[testnum][0],
+                                 seconds.kem);
```

Comment:
nit: indentation is off by one

Reply:
corrected in https://github.com/openssl/openssl/pull/19968/commits/7b25a24f195e11f45dd2d8c1f94e80fde36d4360